frontend/src/RecordingPageComp/PlaybackContainerComp.jsx

```diff
@@ -1,69 +1,273 @@
-            const input = buffer.getChannelData(0); // mono audio
-            const detector = PitchDetector.forFloat32Array(input.length);
-            const [pitch] = detector.findPitch(
-              input,
-              audioContextRef.current.sampleRate
-            );
-            frequencyToMIDINoteNumber(pitch);
-            setPlaybackDuration(buffer.duration);
-          });
+          audioContextRef.current.decodeAudioData(
+            arrayBuffer,
+            (buffer) => {
+              let input = buffer.getChannelData(0); // mono audio
+
+              // Pre-processing: Simple normalization
+
+              let maxVal = 0;
+              for (let i = 0; i < input.length; i++) {
+                const absValue = Math.abs(input[i]);
+                if (absValue > maxVal) {
+                  maxVal = absValue;
+                }
+              }
+              if (maxVal > 0) {
+                input = input.map((x) => x / maxVal);
+              }
+
+              //Multipitch Detector
+              const windowSize = 1024;
+              const hopSize = 512;
+              const pitches = analyzePitchOverTime(
+                input,
+                sampleRate,
+                windowSize,
+                hopSize
+              );
+
+              const groupedPitches = groupPitches(pitches, sampleRate, hopSize);
+              const filteredPitches = filterShortDurations(groupedPitches, 0.1); // Filter out short durations
+              setMidiNotes(filteredPitches);
+
+              //Single Pitch detector
+              const singlePitchDetector = PitchDetector.forFloat32Array(
+                input.length
+              );
+              const [pitch] = singlePitchDetector.findPitch(
+                input,
+                audioContextRef.current.sampleRate
+              );
+              //Values will be passed on to SimilarSounds
+              setNote(frequencyToMIDINoteNumber(pitch));
+              setPlaybackDuration(buffer.duration);
+            },
+            (error) => {
+              console.error("Error decoding audio data:", error);
+            }
+          );
         })
         .catch((error) => {
           console.error("Error reading audio blob:", error);
         });
     }
   };
 
-  const playTestInstrument = () => {
-    midiSounds.current.playChordNow(
-      selectedInstrument,
-      [note],
-      playbackDuration
-    );
-  };
+  function analyzePitchOverTime(buffer, sampleRate, windowSize, hopSize) {
+    const pitches = [];
+
+    for (let start = 0; start + windowSize <= buffer.length; start += hopSize) {
+      let windowBuffer = buffer.slice(start, start + windowSize);
+      const pitch = yin(windowBuffer, sampleRate);
+      if (pitch) {
+        const midiNote = frequencyToMIDINoteNumber(pitch);
+        pitches.push(midiNote);
+      } else {
+        pitches.push(null);
+      }
+    }
+    return pitches;
+  }
+
+  function yin(buffer, sampleRate) {
```

Comment:
I would either have a comment explaining what this algorithm does or rename it to describe what it does